akka-contrib/src/main/scala/akka/contrib/pattern/ReceivePipeline.scala

```diff
@@ -38,10 +38,13 @@ trait ReceivePipeline extends Actor {
     val around = aroundCache match {
       case Some((`receive`, cached)) ⇒ cached
       case _ ⇒
-        val zipped = pipeline.foldRight[Receive](receive)((outer, inner) ⇒ outer(inner).orElse[Any, Unit](inner))
+        val zipped = pipeline.foldRight[Receive] {
+          case msg => super.aroundReceive(receive, msg)
```

Comment:
You're right @ktoso, in a mixin context ```super.aroundReceive``` might not be just the default impl, and if any of the interceptors intentionally discard messages ```super.aroundReceive``` won't be called.

We should still use the original ```receive``` as zero but with fallback in ```unhandled```. ```receive.orElse(PartialFunction(unhandled))```

I saw that this was your first approach for fixing it @jeremystone. What was wrong with it that you changed it?